### ISSUE TITLE
Chat history usage is 4x times bigger than the configured --max-chat-history-tokens

### ISSUE THREAD

**Issue by distributev** [Aider-AI/aider#2003]
### Issue

Hello,

I re-opened an older issue because I'm not sure that this is expected.

Chat history is the biggest contributor to my Claude Sonnet costs.

You can find a screenshot in the below issue.

https://github.com/Aider-AI/aider/issues/1963

### Version and model info

Latest aider with Claude Sonnet.

**Comment by fry69** [comment]
Thank you for filing this issue.

Part of the problem seems to be that you changed the defaults and allowed the chat history grow so big in the first place. The LLM seems to fail to summarize the chat if it gets too large. In doubt use `/clear` the chat history.

A huge chat history is not a benefit, it is ballast. It may contain many outdated ideas and rejected code paths. It will only confuse the LLM, not help it. Please try to use `/clear` as often as possible.

**Comment by distributev** [comment]
Hello,

Which defaults are you referring when you say "you changed the defaults"? I did not change any default I just used --max-chat-history-tokens hoping to limit the chat history and to avoid exactly what you are saying "allow the chat history grow so big"

But, as visible in the screenshot, after I used --max-chat-history-tokens hoping to limit the history the chat history still became 4x bigger vs. what I configured.

**Comment by distributev** [comment]
Actually I don't care if I do --max-chat-history-tokens or not.

What I care is to limit the size up to which the chat history is growing.

How I can do that?

**Comment by fry69** [comment]
> What I care is to limit the size up to which the chat history is growing.
> 
> How I can do that?

Use `/tokens` and `/clear` regularly.

**Comment by distributev** [comment]
Use /tokens and /clear regularly - This I do but I forgot sometimes and it breaks the flow to keep remember to /clear
because of that I looked for something like --max-chat-history-token to limit the history. Actually I noticed that many times even when there is not history at all aider responses are very good sometimes better than when there is a lot of history.

Because of that I was thinking "If I can limit the history to stop growing" it will be perfect I don't ever need to be worry about the history (most of the times the history is the single most expensive part of the costs more than the files I manually add with /add) - I do /reset and /clear but ... if --max-chat-history-token would work it would be perfect.
